blockchain/types/derivesha/mux.go

```diff
@@ -33,8 +33,13 @@ type IDeriveSha interface {
 	DeriveSha(list types.DerivableList) common.Hash
 }
 
+type GovernanceEngine interface {
+	ParamsAt(num uint64) (*params.GovParamSet, error)
+}
+
 var (
 	config     *params.ChainConfig
+	gov        GovernanceEngine
 	instances  map[int]IDeriveSha
```

Comment:
Good suggestion. Since it is already in `derivesha` package, I'd like to omit the deriveSha- prefix.
@hyunsooda How about calling it `implementations` or `impls` instead?

Reply:
`impls` seems better than `implementations`, but it's okay to use `implementations` as well.

Reply:
`impl` seems better than `implementations`, but it's okay to use `implementations` as well.

Reply:
Fixed [6054c93](https://github.com/klaytn/klaytn/pull/1708/commits/6054c93d98782e0c4cec6207285b473cf5250525).